src/hub/HubHelpers.sol

```diff
@@ -164,6 +172,48 @@ contract HubHelpers is Auth, IHubHelpers {
         accounting.lock();
     }
 
+    /// @inheritdoc IHubHelpers
+    function request(PoolId poolId, ShareClassId scId, AssetId assetId, bytes calldata payload) external auth {
+        uint8 kind = uint8(RequestMessageLib.requestType(payload));
+
+        if (kind == uint8(RequestType.DepositRequest)) {
+            RequestMessageLib.DepositRequest memory m = payload.deserializeDepositRequest();
+            shareClassManager.requestDeposit(poolId, scId, m.amount, m.investor, assetId);
+        } else if (kind == uint8(RequestType.RedeemRequest)) {
+            RequestMessageLib.RedeemRequest memory m = payload.deserializeRedeemRequest();
+            shareClassManager.requestRedeem(poolId, scId, m.amount, m.investor, assetId);
+        } else if (kind == uint8(RequestType.CancelDepositRequest)) {
+            RequestMessageLib.CancelDepositRequest memory m = payload.deserializeCancelDepositRequest();
+            uint128 cancelledAssetAmount = shareClassManager.cancelDepositRequest(poolId, scId, m.investor, assetId);
+
+            // Cancellation might have been queued such that it will be executed in the future during claiming
+            if (cancelledAssetAmount > 0) {
+                sender.sendRequestCallback(
+                    poolId,
+                    scId,
+                    assetId,
+                    RequestCallbackMessageLib.FulfilledDepositRequest(m.investor, 0, 0, cancelledAssetAmount).serialize(
+                    )
+                );
+            }
+        } else if (kind == uint8(RequestType.CancelRedeemRequest)) {
+            RequestMessageLib.CancelRedeemRequest memory m = payload.deserializeCancelRedeemRequest();
+            uint128 cancelledShareAmount = shareClassManager.cancelRedeemRequest(poolId, scId, m.investor, assetId);
+
+            // Cancellation might have been queued such that it will be executed in the future during claiming
+            if (cancelledShareAmount > 0) {
+                sender.sendRequestCallback(
+                    poolId,
+                    scId,
+                    assetId,
+                    RequestCallbackMessageLib.FulfilledRedeemRequest(m.investor, 0, 0, cancelledShareAmount).serialize()
+                );
+            }
+        } else {
+            revert UnknownRequestType();
+        }
+    }
```

Comment:
@hieronx why is this in the helpers? Looks like it should go into the core.

Reply:
In general I think in v3.1, we want to find a way to refactor `Hub` and `HubHelpers` somehow. Potentially just moving all incoming gateway actions.